empiric-package/empiric/cli/contracts/oracle.py

```diff
@@ -85,12 +85,23 @@ async def publish_entry(entry: str, config_path=config.DEFAULT_CONFIG):
 
 @app.command()
 @coro
-async def cp(pair_id: str, config_path=config.DEFAULT_CONFIG):
+async def cp(pair_id: str, config_path=config.DEFAULT_CONFIG, max_fee = DEFAULT_MAX_FEE, estimate_fee: bool = False):
+
     client = net.init_empiric_client(config_path)
-    invocation = await client.oracle.set_checkpoint.invoke(
-        str_to_felt(pair_id),
-        0,
-        max_fee=DEFAULT_MAX_FEE,
+    max_fee_to_use = max_fee
+    prepared_call = client.oracle.set_checkpoint.prepare(str_to_felt(pair_id),
+                    0,
+        )
+ 
+    if estimate_fee == True:
+        fee =  await prepared_call.estimate_fee()
```

Comment:
very cool, so you can just prepare the invocation and then call estimate fee rather than invoke, which I believe is equivalent to `prepare(*args, **kwargs).invoke()`